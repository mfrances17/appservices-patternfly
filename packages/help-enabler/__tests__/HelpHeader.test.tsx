import React, { useContext } from "react";
import { render, act, fireEvent } from "@testing-library/react";
import { HelpHeader, HelpContext } from "../HelpHeader";

describe("<HelpHeader />", () => {
  it("render", () => {
    const comp = render(<HelpHeader />);
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it("open dropdown", () => {
    const comp = render(<HelpHeader />);
    const button = document.querySelector("button");
    act(() => {
      fireEvent.click(button!);
    });
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it("enable help", () => {
    const HelpEnabled = () => {
      const { enabled } = useContext(HelpContext);
      return <div id="result">{enabled ? "YES" : "NO"}</div>;
    };
    const comp = render(
      <>
        <HelpHeader />
        <HelpEnabled />
      </>
    );

    const button = document.querySelector("button");
    act(() => {
      fireEvent.click(button!);
    });

    const switchComp = document.querySelector("span.pf-c-switch__toggle");
    act(() => {
      fireEvent.click(switchComp!);
    });

    expect(comp.asFragment()).toMatchSnapshot();
    expect(document.getElementById("result")?.innerHTML).toBe("YES");
  });
});
