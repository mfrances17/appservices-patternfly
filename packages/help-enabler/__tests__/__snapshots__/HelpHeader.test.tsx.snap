// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelpHeader /> enable help 1`] = `
<DocumentFragment>
  <div
    class="pf-c-dropdown pf-m-align-right pf-m-expanded"
    data-ouia-component-id="OUIA-Generated-Dropdown-3"
    data-ouia-component-type="PF4/Dropdown"
    data-ouia-safe="true"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      aria-label="Help"
      class="pf-c-dropdown__toggle pf-m-plain"
      id="help"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0"
          />
        </svg>
      </span>
    </button>
    <ul
      aria-labelledby="help"
      class="pf-c-dropdown__menu pf-m-align-right"
      role="menu"
    >
      <li
        id="link"
        role="menuitem"
      >
        <a
          aria-disabled="false"
          class="pf-c-dropdown__menu-item"
          tabindex="-1"
        >
          <div
            class="pf-l-split"
          >
            <div
              class="pf-l-split__item pf-m-fill"
            >
              documentation
            </div>
            <div
              class="pf-l-split__item"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                />
              </svg>
            </div>
          </div>
        </a>
      </li>
      <li
        id="enable"
        role="menuitem"
      >
        <a
          aria-disabled="false"
          class="pf-c-dropdown__menu-item pf-m-description"
          tabindex="-1"
        >
          <div
            class="pf-c-dropdown__menu-item-main"
          >
            <div
              class="pf-l-split"
            >
              <div
                class="pf-l-split__item pf-m-fill"
              >
                enableHelpMode
              </div>
              <div
                class="pf-l-split__item"
              >
                <label
                  class="pf-c-switch"
                  data-ouia-component-id="OUIA-Generated-Switch-2"
                  data-ouia-component-type="PF4/Switch"
                  data-ouia-safe="true"
                  for="enableHelp"
                >
                  <input
                    aria-label="Help is enabled"
                    checked=""
                    class="pf-c-switch__input"
                    id="enableHelp"
                    type="checkbox"
                  />
                  <span
                    class="pf-c-switch__toggle"
                  >
                    <div
                      aria-hidden="true"
                      class="pf-c-switch__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 512 512"
                        width="1em"
                      >
                        <path
                          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                        />
                      </svg>
                    </div>
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            class="pf-c-dropdown__menu-item-description"
          >
            This toggle will enable / disable part of the help info in the console. Includes any help text, links and popovers.
          </div>
        </a>
      </li>
    </ul>
  </div>
  <div
    id="result"
  >
    YES
  </div>
</DocumentFragment>
`;

exports[`<HelpHeader /> open dropdown 1`] = `
<DocumentFragment>
  <div
    class="pf-c-dropdown pf-m-align-right pf-m-expanded"
    data-ouia-component-id="OUIA-Generated-Dropdown-2"
    data-ouia-component-type="PF4/Dropdown"
    data-ouia-safe="true"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      aria-label="Help"
      class="pf-c-dropdown__toggle pf-m-plain"
      id="help"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0"
          />
        </svg>
      </span>
    </button>
    <ul
      aria-labelledby="help"
      class="pf-c-dropdown__menu pf-m-align-right"
      role="menu"
    >
      <li
        id="link"
        role="menuitem"
      >
        <a
          aria-disabled="false"
          class="pf-c-dropdown__menu-item"
          tabindex="-1"
        >
          <div
            class="pf-l-split"
          >
            <div
              class="pf-l-split__item pf-m-fill"
            >
              documentation
            </div>
            <div
              class="pf-l-split__item"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                />
              </svg>
            </div>
          </div>
        </a>
      </li>
      <li
        id="enable"
        role="menuitem"
      >
        <a
          aria-disabled="false"
          class="pf-c-dropdown__menu-item pf-m-description"
          tabindex="-1"
        >
          <div
            class="pf-c-dropdown__menu-item-main"
          >
            <div
              class="pf-l-split"
            >
              <div
                class="pf-l-split__item pf-m-fill"
              >
                enableHelpMode
              </div>
              <div
                class="pf-l-split__item"
              >
                <label
                  class="pf-c-switch"
                  data-ouia-component-id="OUIA-Generated-Switch-1"
                  data-ouia-component-type="PF4/Switch"
                  data-ouia-safe="true"
                  for="enableHelp"
                >
                  <input
                    aria-label="Help is enabled"
                    checked=""
                    class="pf-c-switch__input"
                    id="enableHelp"
                    type="checkbox"
                  />
                  <span
                    class="pf-c-switch__toggle"
                  >
                    <div
                      aria-hidden="true"
                      class="pf-c-switch__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 512 512"
                        width="1em"
                      >
                        <path
                          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                        />
                      </svg>
                    </div>
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            class="pf-c-dropdown__menu-item-description"
          >
            This toggle will enable / disable part of the help info in the console. Includes any help text, links and popovers.
          </div>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<HelpHeader /> render 1`] = `
<DocumentFragment>
  <div
    class="pf-c-dropdown pf-m-align-right"
    data-ouia-component-id="OUIA-Generated-Dropdown-1"
    data-ouia-component-type="PF4/Dropdown"
    data-ouia-safe="true"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      aria-label="Help"
      class="pf-c-dropdown__toggle pf-m-plain"
      id="help"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
